app/models/certificate_service.rb

```diff
@@ -4,4 +4,27 @@ class CertificateService < ActiveRecord::Base
   belongs_to :service, polymorphic: true
 
   validates :certificate, uniqueness: { scope: [:service_id, :service_type] }
+
+  after_create :update_service_id
+  before_destroy :nullify_service_reference
+
+  private
+
+  def service_type_instance
+    service_type.constantize.find(service_id)
+  end
+
+  def update_service_id
```

Comment:
Is this necessary? When we create a model I think Rails is smart enough to associate all the needed ids, no? Which scenario is this needed?

Reply:
unfortunately not, could be that it can be done with some model attribute/configuration but i couldnt find anything

Reply:
Really? Weird...

Reply:
I don't see why we need to do this. In fact, I don't understand why we need the `certificate_service_id` attribute on `Registry` or `RegistryMirror`, we should drop them. This is why this is messing up with us. The association is smart enough to use the proxy model.